From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@microsoft.com>
Date: Mon, 31 Jan 2022 15:48:37 -0600
Subject: [PATCH] cmd/dist: add explicit default timeout for cgo

Add "t.timeout" call to cgo test command with the default value, 600.
This makes GO_TEST_TIMEOUT_SCALE have an effect on cgo test timeout.
---
 src/cmd/dist/test.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cmd/dist/test.go b/src/cmd/dist/test.go
index 518b35c53a..b63bf1fcdc 100644
--- a/src/cmd/dist/test.go
+++ b/src/cmd/dist/test.go
@@ -1136,7 +1136,7 @@ func (t *tester) runHostTest(dir, pkg string) error {
 }
 
 func (t *tester) cgoTest(dt *distTest) error {
-	cmd := t.addCmd(dt, "misc/cgo/test", t.goTest())
+	cmd := t.addCmd(dt, "misc/cgo/test", t.goTest(), t.timeout(600))
 	setEnv(cmd, "GOFLAGS", "-ldflags=-linkmode=auto")
 
 	// Skip internal linking cases on linux/arm64 to support GCC-9.4 and above.
